src/selectors/index.ts

```diff
@@ -31,6 +46,17 @@ export const getFilters = (state: State) => state.present.shelf.spec.filters;
 export const getShelf = (state: State) => state.present.shelf;
 export const getSchema = (state: State) => state.present.dataset.schema;
 export const getMainResult = (state: State) => state.present.result.main.modelGroup;
+export const getSpec = (state: State) => {
```

Comment:
1) You should use `createSelector` to correctly chain the dependencies.
(Maybe this is why it does not solve the problem yet.) 

2) rename this `getMainSpec` and make it depends on `query` as well.